solver/src/solver/http_solver.rs

```diff
@@ -145,16 +145,15 @@ impl HttpSolver {
     fn token_models(
         &self,
         token_infos: &HashMap<H160, TokenInfo>,
-        price_estimates: &HashMap<H160, Result<BigRational, PriceEstimationError>>,
+        price_estimates: &HashMap<H160, f64>,
         buffers: &HashMap<H160, U256>,
     ) -> HashMap<H160, TokenInfoModel> {
         token_infos
             .iter()
             .map(|(address, token_info)| {
-                let external_price = price_estimates[address]
-                    .as_ref()
-                    .ok()
-                    .and_then(|price| price.to_f64());
+                // TODO: Do we need to filter out infinity? This can happen when buy amount is 0 so
+                // sell / buy becomes infinity.
+                let external_price = price_estimates.get(address).copied();
```

Comment:
Question for @marcovc .

Reply:
You mean that selling about 1 ETH of a token would return 0 of another token?

Reply:
Yes. Would be easy to filter out just not sure if we need to. I guess similar question for a 0 price even though I don't think it can happen.

Reply:
Just for my understanding: I've not seen the current code, but before this PR the price passed to the solver was simply cast from a rational to float, wasn't it? And since a rational can't have zero denominator, then why we need to make this decision now?

Reply:
correct

Reply:
I just saw that your last sentence was a question. The reason this wasn't relevant before is that an infinite price would have been filtered out previously as it cannot be represented by a BigRational. Now we do not go through Rational so we can have a float here that was divided by 0 which becomes infinity.

Reply:
Ok, I guess we should proceed as we were before for (Rationals with zero denominators) and just don't include that external price. This will work for the http solvers (as we don't depend on these external prices) but I guess it might fail for the ranking functionality in the driver (if there is no price for a traded token we can't score the solution).

Reply:
I will keep the behavior from before. But it sounds like we should address the case where we cannot create a token model or where the token model wouldn't have a price. This is an issue without this PR too because we don't remove orders without token model or price.

Reply:
Actually I see external_price is Optional in the model so we can continue to turn infinite prices into None.